### DISCUSSION TITLE
Issue with Shelly 2.5 (on latest Tasmota firmware) with button switch & control light

### DISCUSSION THREAD

**Discussion by ghost** [arendst/Tasmota#14476] [Q&A] [answered]
### PROBLEM DESCRIPTION
I use a shelly 2.5 with latest tasmota firmware and a push button with control light (AS Schwabe). The control light I need so somebody will be able to find the switch when it is dark.

I use the following button (configured as button NOT switch):
![image](https://user-images.githubusercontent.com/33456/149666212-de3cc562-bd1a-4e51-a305-fecfc33966ca.png)

Connecting the button:
![image](https://user-images.githubusercontent.com/33456/149666510-c79056a8-63a3-4c1b-8f00-0feb7bbbeaf1.png)

My minimal configuration to reproduce the issue:
![image](https://user-images.githubusercontent.com/33456/149666546-0cf83da1-f2ea-4950-8083-167a69bf62a5.png)

Everything works fine without the control light. Once I connect the neutral (blue) wire the control light turns on but the button stops working and I can no longer use it to operate the light.
I assume this is because shelly 2.5 makes the sw1 connector high and when I connect the blue wire it stops the pin from working correctly. I am convinced this is not a Tasmota issue per se but I hope this can be fixed using Tasmota or a workaround can be found.   I have not yet come up with a solution so I hope someone here can help me out.
Note: I previously used a Sonoff mini but this uses the button to pull the sw1 pin to GND (not compatible with my button using the control light) so I moved to Shelly 2.5.

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [x] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [x] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [x] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [x] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [x] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [x] Device used (e.g., Sonoff Basic): Shelly 2.5
- [x] Tasmota binary firmware version number used: Core/SDK Version 2_7_4_9/2.2.2-dev(38a443e)
  - [x] Pre-compiled
  - [ ] Self-compiled
- [x] Flashing tools used: Tasmota Web
- [x] Provide the output of command: `Backlog Template; Module; GPIO 255`:
```lua
  Configuration output here:
16:48:29.669 CMD: Backlog Template; Module; GPIO 255
16:48:29.693 MQT: stat/tasmota_8DCED6/RESULT = {"NAME":"Shelly 2.5","GPIO":[320,0,32,0,224,193,0,0,640,192,608,225,3456,4736],"FLAG":0,"BASE":18}
16:48:29.916 MQT: stat/tasmota_8DCED6/RESULT = {"Module":{"0":"Shelly 2.5"}}
16:48:30.124 MQT: stat/tasmota_8DCED6/RESULT = {"GPIO0":{"320":"Led_i1"},"GPIO1":{"0":"None"},"GPIO2":{"32":"Button1"},"GPIO3":{"0":"None"},"GPIO4":{"224":"Relay1"},"GPIO5":{"193":"Switch_n2"},"GPIO9":{"0":"None"},"GPIO10":{"0":"None"},"GPIO12":{"640":"I2C SDA"},"GPIO13":{"192":"Switch_n1"},"GPIO14":{"608":"I2C SCL"},"GPIO15":{"225":"Relay2"},"GPIO16":{"3456":"ADE7953 IRQ1"},"GPIO17":{"4736":"ADC Temp"}}

```
- [ ] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```lua
  Rules output here:


```
- [x] Provide the output of this command: `Status 0`:
```lua
  STATUS 0 output here:
16:47:36.991 CMD: status 0
16:47:36.998 MQT: stat/tasmota_8DCED6/STATUS = {"Status":{"Module":0,"DeviceName":"Tasmota","FriendlyName":["Tasmota",""],"Topic":"tasmota_8DCED6","ButtonTopic":"0","Power":0,"PowerOnState":3,"LedState":1,"LedMask":"FFFF","SaveData":1,"SaveState":1,"SwitchTopic":"0","SwitchMode":[0,0,0,0,0,0,0,0],"ButtonRetain":0,"SwitchRetain":0,"SensorRetain":0,"PowerRetain":0,"InfoRetain":0,"StateRetain":0}}
16:47:37.002 MQT: stat/tasmota_8DCED6/STATUS1 = {"StatusPRM":{"Baudrate":115200,"SerialConfig":"8N1","GroupTopic":"tasmotas","OtaUrl":"http://ota.tasmota.com/tasmota/release/tasmota.bin.gz","RestartReason":"External System","Uptime":"0T00:00:30","StartupUTC":"2022-01-16T15:47:06","Sleep":50,"CfgHolder":4617,"BootCount":9,"BCResetTime":"2022-01-16T15:53:53","SaveCount":44,"SaveAddress":"F8000"}}
16:47:37.037 MQT: stat/tasmota_8DCED6/STATUS2 = {"StatusFWR":{"Version":"10.1.0(tasmota)","BuildDateTime":"2022-01-09T14:55:09","Boot":31,"Core":"2_7_4_9","SDK":"2.2.2-dev(38a443e)","CpuFrequency":80,"Hardware":"ESP8266EX","CR":"375/699"}}
16:47:37.057 MQT: stat/tasmota_8DCED6/STATUS3 = {"StatusLOG":{"SerialLog":2,"WebLog":2,"MqttLog":0,"SysLog":0,"LogHost":"","LogPort":514,"SSId":["LAFERME",""],"TelePeriod":300,"Resolution":"558180C0","SetOption":["00008009","2805C80001000600003C5A0A000000000000","00000080","00006000","00004000"]}}
16:47:37.090 MQT: stat/tasmota_8DCED6/STATUS4 = {"StatusMEM":{"ProgramSize":616,"Free":384,"Heap":24,"ProgramFlashSize":1024,"FlashSize":2048,"FlashChipId":"15701C","FlashFrequency":40,"FlashMode":3,"Features":["00000809","8FDAC787","04368001","000000CF","010013C0","C000F981","00004004","00001000","00000020"],"Drivers":"1,2,3,4,5,6,7,8,9,10,12,16,18,19,20,21,22,24,26,27,29,30,35,37,45","Sensors":"1,2,3,4,5,6"}}
16:47:37.123 MQT: stat/tasmota_8DCED6/STATUS5 = {"StatusNET":{"Hostname":"tasmota-8DCED6-3798","IPAddress":"172.16.0.38","Gateway":"172.16.0.1","Subnetmask":"255.255.255.0","DNSServer1":"192.168.2.1","DNSServer2":"0.0.0.0","Mac":"48:3F:DA:8D:CE:D6","Webserver":2,"HTTP_API":1,"WifiConfig":4,"WifiPower":17.0}}
16:47:37.152 MQT: stat/tasmota_8DCED6/STATUS6 = {"StatusMQT":{"MqttHost":"172.16.0.12","MqttPort":1883,"MqttClientMask":"DVES_%06X","MqttClient":"DVES_8DCED6","MqttUser":"mqtt","MqttCount":1,"MAX_PACKET_SIZE":1200,"KEEPALIVE":30,"SOCKET_TIMEOUT":4}}
16:47:37.174 MQT: stat/tasmota_8DCED6/STATUS7 = {"StatusTIM":{"UTC":"2022-01-16T15:47:37","Local":"2022-01-16T16:47:37","StartDST":"2022-03-27T02:00:00","EndDST":"2022-10-30T03:00:00","Timezone":"+01:00","Sunrise":"08:37","Sunset":"17:21"}}
16:47:37.192 MQT: stat/tasmota_8DCED6/STATUS9 = {"StatusPTH":{"PowerDelta":[0,0,0],"PowerLow":0,"PowerHigh":0,"VoltageLow":0,"VoltageHigh":0,"CurrentLow":0,"CurrentHigh":0}}
16:47:37.213 MQT: stat/tasmota_8DCED6/STATUS10 = {"StatusSNS":{"Time":"2022-01-16T16:47:37","Switch1":"ON","Switch2":"OFF","ANALOG":{"Temperature":33.5},"ENERGY":{"TotalStartTime":"2022-01-16T15:53:53","Total":0.000,"Yesterday":0.000,"Today":0.000,"Power":[ 0, 0],"ApparentPower":[ 0, 0],"ReactivePower":[ 0, 0],"Factor":[0.00,0.00],"Voltage": 0,"Current":[0.000,0.000]},"TempUnit":"C"}}
16:47:37.248 MQT: stat/tasmota_8DCED6/STATUS11 = {"StatusSTS":{"Time":"2022-01-16T16:47:37","Uptime":"0T00:00:31","UptimeSec":31,"Heap":24,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":1,"POWER1":"OFF","POWER2":"OFF","Wifi":{"AP":1,"SSId":"LAFERME","BSSId":"E0:63:DA:B7:D8:49","Channel":1,"Mode":"11n","RSSI":88,"Signal":-56,"LinkCount":1,"Downtime":"0T00:00:03"}}}

```
- [x] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
```lua
  Console output here:
16:50:18.409 ADE: 5972094, 4473, [1773, -34, 630, 81], [1773, 0, 630, -12]
16:50:18.411 ADE: U 5972094, C 4473, I 0 + 0 = 0, P 0 + 0
16:50:19.409 ADE: 5970484, 4474, [1773, -57, 630, 43], [1773, -7, 630, 14]
16:50:19.411 ADE: U 5970484, C 4474, I 0 + 0 = 0, P 0 + 0
16:50:20.410 ADE: 5972256, 4474, [1773, -28, 630, 30], [1773, -12, 630, 3]
16:50:20.412 ADE: U 5972256, C 4474, I 0 + 0 = 0, P 0 + 0
16:50:21.410 ADE: 5981165, 4474, [1773, -76, 631, 127], [1773, -5, 631, 4]
16:50:21.412 ADE: U 5981165, C 4474, I 0 + 0 = 0, P 0 + 0
16:50:22.410 ADE: 5981451, 4474, [1773, -41, 631, 59], [1773, 7, 631, 3]
16:50:22.413 ADE: U 5981451, C 4474, I 0 + 0 = 0, P 0 + 0
16:50:23.410 ADE: 5982618, 4474, [1773, -49, 631, 32], [1773, 5, 631, 3]
16:50:23.412 ADE: U 5982618, C 4474, I 0 + 0 = 0, P 0 + 0
16:50:24.410 ADE: 5967093, 4475, [1773, -97, 630, 68], [1773, 4, 630, -3]
16:50:24.413 ADE: U 5967093, C 4475, I 0 + 0 = 0, P 0 + 0
16:50:25.409 ADE: 5972912, 4474, [1773, -12, 630, 76], [1773, 23, 630, 14]
16:50:25.411 ADE: U 5972912, C 4474, I 0 + 0 = 0, P 0 + 0
16:50:26.410 ADE: 5972811, 4475, [1773, 14, 630, 68], [1773, -26, 630, 17]
16:50:26.413 ADE: U 5972811, C 4475, I 0 + 0 = 0, P 0 + 0
16:50:27.410 ADE: 5966208, 4475, [1773, -74, 630, 27], [1773, -13, 629, 19]
16:50:27.412 ADE: U 5966208, C 4475, I 0 + 0 = 0, P 0 + 0
16:50:28.410 ADE: 5976500, 4474, [1773, -20, 631, 14], [1773, -16, 631, -17]
16:50:28.412 ADE: U 5976500, C 4474, I 0 + 0 = 0, P 0 + 0

```

### TO REPRODUCE
_Steps to reproduce the behavior:_
see problem description above

### EXPECTED BEHAVIOUR
_A clear and concise description of what you expected to happen._
see problem description above

### SCREENSHOTS
_If applicable, add screenshots to help explain your problem._

### ADDITIONAL CONTEXT
_Add any other context about the problem here._

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by barbudor** [comment]
@markfink, please add a diagram of your cabling. That will be more easy to read than a photo. Thanks

**Reply by ghost** [reply]
note: I currently use only channel 1 but plan to extend to channel 2 in case it works...
![cabling diagram for shelly 2.5](https://user-images.githubusercontent.com/33456/149815416-a004b62b-31cf-44c4-bf45-8affa46b7f8f.png)

**Comment by barbudor** [comment]
By the way, why changing the external input from Switch to Button ?
You can configure a Switch as a push-button using `Switchmode 4`
Changing the external from Switch to Button have some side-effects as it brings to the external button some features that are specific:
- double press with toggle relay 2
- long press >40 will trigger recovery mode (unless disabled by appropriate setoption)

Of course, you can see some advantage to bring those from the un-reachable on-device button to an external one.

Looking forward your explanation of what you want to achieve with the integrated light

**Reply by ghost** [reply]
The configuration I have is not possible with switches. Like I said above I made a small experiment to show the problem. The following diagram shows how I want my  configuration of multiple illuminated buttons (currently 4x buttons but I plan to extend) to control multiple lights (currently 5x 1.2m LED 36w) on / off but with WIFI: 
https://upload.wikimedia.org/wikipedia/commons/1/10/Colour-wiring_sw_cross_switch.svg

To my knowledge this is today solved with pushbuttons since the wiring can be simpler and the pushbuttons are 10x cheaper components than double switches. AFAIK there are today no more illuminated double switches for wet rooms available. Plus I already bought the hardware.  I think my requirements are valid (and yes I use switchmode1 3).

Currently it looks to me like there are no tasmota devices available to handle this job. There might be a lowpass-filter workaround to solve this issue but I am not an expert on these... 
Please correct me if I am wrong but to me it looks like there are only two ways to solve this:
1. buy a std. component that is able to do the job like an Eltako ES61
2. drop the control light requirement (not an option for me since one could not find the button without it at night - I am not sure if people understand what that means but let me vouch for it that you can not see your hand in front of your face! let alone find a button)

I did not expect that a simple control light can eliminate the home automation approach. I checked that everything works in advance but only with a std. switch. Now I know better...

**Reply by barbudor** [reply]
First, do understand there is a difference between a physical button / switch and a Tasmota configuration Button / Switch
While a Tasmota Button can be connected only to a push-button, a Tasmota Switch item can be used with either a push-button or a rocker-switch.
As I understand you want to use physical push-buttons, it doesn't mean you must use a Tasmota Button item. A Tasmota Switch can be used with a push-button, all is about the SwitchMode. And a physical push-button used as a Button or used as a Switch in SwitchMode 4 have different behavior in Tasmota. Depending on your choice, one may be more interresting than the other.
But we can move on with physical push-button and a Tasmota Button configuration.

Then, don't try to replicate a circuit that is using dumb switches using smart switches.
You only need 1 Shelly with 1 channel to actively control the light, and it doesn't even need to be located with the buttons (ok, may be at the position of the last one)


So let me recap my understanding 
- You have 1 light to control
- You want to control the light from 3 different locations using push-buttons

Questions:
- Is the circuit using dumb switches already in place exactly as on the diagram ?
- Which means there is no neutral at the position of the switches/buttons ?

**Reply by ghost** [reply]
to answer your recap paste from above:

- You have 1 light to control
- You want to control the light from 3 different locations using push-buttons

`want my configuration of multiple illuminated buttons (currently 4x buttons but I plan to extend) to control multiple lights (currently 5x 1.2m LED 36w) on / off 
`

to answer your question paste from above:

- Is the circuit using dumb switches already in place exactly as on the diagram ?
- Which means there is no neutral at the position of the switches/buttons ?

`Everything works fine without the control light. Once I connect the neutral (blue) wire the control light turns on but the button stops working and I can no longer use it to operate the light.
I assume this is because shelly 2.5 makes the sw1 connector high and when I connect the blue wire it stops the pin from working correctly. I am convinced this is not a Tasmota issue per se but I hope this can be fixed using Tasmota or a workaround can be found. I have not yet come up with a solution so I hope someone here can help me out.
Note: I previously used a Sonoff mini but this uses the button to pull the sw1 pin to GND (not compatible with my button using the control light) so I moved to Shelly 2.5.`

**Comment by barbudor** [comment, marked as answer]
The diagram below should allow you achieve what you need by re-using the wires that are already in place following the diagram you sent.
Except that for the small light in the button, you must have the neutral also at the location of the button.
Trying to connect the button's light between the Line and the SW1 may at best not work, at worst destroy the SW input of the Shelly

![image](https://user-images.githubusercontent.com/2996042/149828264-f936f453-70be-4497-914c-e2e83ab76078.png)

**Reply by ghost** [reply]
the push button I use is a std component (see above). Unfortunately I was not able to get a diagram for it so I did some "reverse engineering" and what I found is IMHO how at least a few of similar components are manufactured:
![shelly_diagram_v1](https://user-images.githubusercontent.com/33456/149836276-28d7ad19-3816-4a1f-ad63-83951cb94144.png)

**Reply by ascillato** [reply]
Are you sure that your push button connects N to SW1 and when you push it connects L to SW1 ? that is weird. if it does that, your shelly is seeing Mains AC on SW1 when you push and also when you release. Shelly don't differentiate if it is L or N what you put in the SW1 input.

**Reply by ghost** [reply]
What I want does not work like that... that I am sure of. Like I said above it works until I connect the N wire to "g". 
To me it looks like the way I connect my "toggle devices" can not be handled by shelly or any other currently available tasmota capable ESP8266 devices or ESP32.

**Reply by barbudor** [reply]
Connecting SW1 to N is WRONG

**Reply by ghost** [reply]
I understand now that the push button wiring I use is wrong and the Eltako can not fix it. I try to come up with a better solution and report back.

**Reply by ghost** [reply]
@barbudor thank you so much for your help. I now understand that I bought the wrong push buttons for the job (I returned them and bought suitable ones). I am just a weekend warrior and I had some trouble finding what is causing the issue. You helped me to get unstuck!
Now I go bury the shelly that did not survive the smoke test :)
